tests/test_scalars.py

```diff
@@ -403,261 +454,485 @@ def test_timedelta_cmp() -> None:
+        assert_type(td != c_ndarray_td64, np_ndarray_bool), np.ndarray, np.bool_
+    )
     assert (eq_a != ne_a).all()
 
-    eq_a = check(assert_type(td == c5, np_ndarray_bool), np.ndarray)
-    ne_a = check(assert_type(td != c5, np_ndarray_bool), np.ndarray)
+    eq_a = check(
+        assert_type(td == c_timedelta_index, np_ndarray_bool), np.ndarray, np.bool_
+    )
+    ne_a = check(
+        assert_type(td != c_timedelta_index, np_ndarray_bool), np.ndarray, np.bool_
+    )
     assert (eq_a != ne_a).all()
 
-    eq_s = check(assert_type(td == c6, "pd.Series[bool]"), pd.Series, bool)
-    ne_s = check(assert_type(td != c6, "pd.Series[bool]"), pd.Series, bool)
+    eq_s = check(
+        assert_type(td == c_timedelta_series, "pd.Series[bool]"), pd.Series, bool
+    )
+    ne_s = check(
+        assert_type(td != c_timedelta_series, "pd.Series[bool]"), pd.Series, bool
+    )
     assert (eq_s != ne_s).all()
 
-    eq = check(assert_type(c2 == td, bool), bool)
-    ne = check(assert_type(c2 != td, bool), bool)
+    eq = check(assert_type(c_dt_timedelta == td, bool), bool)
+    ne = check(assert_type(c_dt_timedelta != td, bool), bool)
     assert eq != ne
 
-    eq = check(assert_type(c3 == td, Any), np.bool_)
-    ne = check(assert_type(c3 != td, Any), np.bool_)
+    eq = check(assert_type(c_timedelta64 == td, Any), np.bool_)
+    ne = check(assert_type(c_timedelta64 != td, Any), np.bool_)
     assert eq != ne
 
-    eq_a = check(assert_type(c4 == td, Any), np.ndarray)
-    ne_a = check(assert_type(c4 != td, Any), np.ndarray)
+    eq_a = check(assert_type(c_ndarray_td64 == td, Any), np.ndarray)
+    ne_a = check(assert_type(c_ndarray_td64 != td, Any), np.ndarray)
     assert (eq_a != ne_a).all()
 
-    eq_a = check(assert_type(c5 == td, np_ndarray_bool), np.ndarray)
-    ne_a = check(assert_type(c5 != td, np_ndarray_bool), np.ndarray)
+    eq_a = check(
+        assert_type(c_timedelta_index == td, np_ndarray_bool), np.ndarray, np.bool_
+    )
+    ne_a = check(
+        assert_type(c_timedelta_index != td, np_ndarray_bool), np.ndarray, np.bool_
+    )
     assert (eq_a != ne_a).all()
 
-    eq_s = check(assert_type(c6 == td, "pd.Series[bool]"), pd.Series, bool)
-    ne_s = check(assert_type(c6 != td, "pd.Series[bool]"), pd.Series, bool)
-    assert (eq_s != ne_s).all()
-
-    eq = check(assert_type(td == 1, bool), bool)
-    ne = check(assert_type(td != 1, bool), bool)
-    assert eq != ne
-
-    eq = check(assert_type(td == (3 + 2j), bool), bool)
-    ne = check(assert_type(td != (3 + 2j), bool), bool)
-    assert eq != ne
-
     eq_s = check(
-        assert_type(td == pd.Series([1, 2, 3]), "pd.Series[bool]"), pd.Series, bool
+        assert_type(c_timedelta_series == td, "pd.Series[bool]"), pd.Series, bool
     )
     ne_s = check(
-        assert_type(td != pd.Series([1, 2, 3]), "pd.Series[bool]"), pd.Series, bool
+        assert_type(c_timedelta_series != td, "pd.Series[bool]"), pd.Series, bool
     )
     assert (eq_s != ne_s).all()
```

Comment:
I don't think we need to check `==` and `!=` when the type we care about, in this case `pd.Timedelta` is the right-hand argument, because with `__eq__` and `__ne__`, the left hand object's method will always be used.  So you could delete the above 5 pairs of tests where `td` is on the right-hand-side of the comparison.

Reply:
Moved to separate function and notes these use the LHS object